src/PKSim.Core/Batch/BatchLogger.cs

```diff
@@ -48,6 +70,53 @@ public void AddParameterValueToDebug(IParameter parameter)
          AddParameterValueToDebug(parameter.Name, parameter.Value);
       }
 
-      public IEnumerable<string> Entries => _entries;
+      public void InitializeWith(StreamWriter streamWriter, NotificationType notificationType)
+      {
+         Clear();
+         _streamWriter = streamWriter;
+         _notificationType = notificationType;
+      }
+
+      public void Clear()
+      {
+         _entries.Clear();
+      }
+   }
+
+   public class BatchLoggerDisposer : IDisposable
+   {
+      private readonly StreamWriter _streamWriter;
+
+      public BatchLoggerDisposer(IBatchLogger batchLogger, string logFilePath, NotificationType notificationType)
+      {
+         _streamWriter = new StreamWriter(logFilePath, append: false);
```

Comment:
Here we decide to use a file but the logger does not know about it. We could in theory write to anything